hoot-core/src/main/cpp/hoot/core/index/metric-hybrid/RFqHybridTree.h

```diff
@@ -207,14 +207,13 @@ class RFqHybridTree : public SubTreeCallback<KeyType, DataType>
    */
   bool _isRLayer(int depth) const
   {
-    bool result;
+    bool result = true;
     // this is the standard case, we haven't exceeded either depth, then alternate layering
     if (depth < _rDepth && depth < _fqDepth)
     {
       result = depth % 2 == 0;
     }
-    // if we've exceeded either depth
-    else if (depth >= _rDepth || depth >= _fqDepth)
+    else  // we've exceeded one depth or the other
```

Comment:
If my Boolean logic doesn't fail me, this includes ALL other options that the original `if` excludes and therefore equals `else`.